sklearn/utils/_indexing.py

```diff
@@ -521,9 +540,29 @@ def resample(*arrays, replace=True, n_samples=None, random_state=None, stratify=
 
     check_consistent_length(*arrays)
 
+    if sample_weight is not None and not replace:
+        raise NotImplementedError(
+            "Resampling with sample_weight is only implemented for replace=True."
+        )
+    if sample_weight is not None and stratify is not None:
+        raise NotImplementedError(
+            "Resampling with sample_weight is only implemented for stratify=None."
+        )
     if stratify is None:
         if replace:
-            indices = random_state.randint(0, n_samples, size=(max_n_samples,))
+            if sample_weight is not None:
+                sample_weight = _check_sample_weight(
+                    sample_weight, first, dtype=np.float64
+                )
+                p = sample_weight / sample_weight.sum()
+            else:
+                p = None
+            indices = random_state.choice(
+                n_samples,
+                size=max_n_samples,
+                p=p,
+                replace=True,
+            )
```

Comment:
I think it will change the results when sample_weight is None. To me it's fine because it doesn't change the statistical properties but we should add an entry in the chanelog, under the changed models section.